torch/_inductor/lowering.py

```diff
@@ -1551,6 +1551,49 @@ def inner_fn(index):
     )
 
 
+@register_lowering(inductor_prims._bucketize, type_promotion_kind=None)
```

Comment:
The [documentation](https://pytorch.org/docs/stable/generated/torch.bucketize.html) for `torch.bucketize` specifies that the `boundaries` "must contain a strictly increasing sequence, or the return value is undefined". In the jagged tensor context, the buckets can be zero-size (corresponding to empty sequences in the jagged tensor). I guess, our lowering here should support zero-sized buckets?

Reply:
Good point, updated the docstring in common.py. In practice it seems like torch.bucketize obeys this behavior and we can define the behavior to work for jagged tensors.